homeassistant/components/sensor/nest.py

```diff
@@ -94,13 +101,20 @@ class NestSensor(Entity):
     def __init__(self, structure, device, variable):
         """Initialize the sensor."""
         self.structure = structure
-        self.device = device
         self.variable = variable
 
-        # device specific
-        self._location = self.device.where
-        self._name = "{} {}".format(self.device.name_long,
-                                    self.variable.replace("_", " "))
+        if device is not None:
+            # device specific
+            self.device = device
+            self._location = self.device.where
+            self._name = "{} {}".format(self.device.name_long,
+                                        self.variable.replace("_", " "))
+        else:
+            # structure only
+            self.device = structure
+            self._name = "{} {}".format(self.structure.name,
```

Comment:
slugify

Reply:
logic here is reverse version of slugify, but I will change "_" to single quote